daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/TransactionVersionTest.scala

```diff
@@ -0,0 +1,297 @@
+            };
+          };
+        }
+      """
+
+//  val (coImplementsPkgId, coImplementsPkg) =
+//    PackageId.assertFromString("coimplements-pkg") -> p"""
+//        module CoImplementsMod {
+//          interface (this: InterfaceCoImplements1) = {
+//            viewtype Unit;
+//            method getPerson: Party;
+//            method getLabel: Text;
+//            choice Destroy (self) (arg: Unit): Unit,
+//              controllers Cons @Party [call_method @'coimplements-pkg':CoImplementsMod:InterfaceCoImplements1 getPerson this] (Nil @Party),
+//              observers Nil @Party
+//              to upure @Unit ();
+//            coimplements 'template-pkg':TemplateMod:Template1 {
+//              view = ();
+//              method getPerson = 'template-pkg':TemplateMod:Template1 {person} this;
+//              method getLabel = 'template-pkg':TemplateMod:Template1 {label} this;
+//            };
+//          };
+//        }
+//       """
+
+  val contractParty: IdString.Party = Ref.Party.assertFromString("contractParty")
+  val templateId: Ref.TypeConName =
+    Ref.TypeConName.assertFromString(s"$implementsPkgId:ImplementsMod:TemplateImplements1")
+  val interfaceId: Ref.TypeConName =
+    Ref.TypeConName.assertFromString(s"$interfacesPkgId:InterfaceMod:Interface1")
+  val contractId: ContractId =
+    Value.ContractId.V1(crypto.Hash.hashPrivateKey("test-transaction-version-contract-id"))
+  val contract: Versioned[Value.ContractInstance] = Versioned(
+    TransactionVersion.assignNodeVersion(newVersion),
+    Value.ContractInstance(
+      templateId,
+      Value.ValueRecord(
+        None,
+        ImmArray(
+          None -> Value.ValueParty(contractParty),
+          None -> Value.ValueText("test"),
+        ),
+      ),
+      "test",
+    ),
+  )
+
+  def evaluateBeginExercise(
+      pkgs: CompiledPackages,
+      templateId: TypeConName,
+      interfaceId: Option[TypeConName],
+      contractId: ContractId,
+      committers: Set[Party] = Set.empty,
+      controllers: Set[Party] = Set.empty,
+      getContract: PartialFunction[Value.ContractId, Value.VersionedContractInstance] =
+        PartialFunction.empty,
+  ): Either[SError.SError, SubmittedTransaction] = {
+    import SpeedyTestLib.loggingContext
+
+    val choiceName = Ref.ChoiceName.assertFromString("Destroy")
+    val choiceArg = SExpr.SEValue(SValue.SUnit)
+    val speedyContractId = SExpr.SEValue(SValue.SContractId(contractId))
+    val speedyControllers =
+      SExpr.SEValue(SValue.SList(FrontStack.from(controllers.map(SValue.SParty))))
+    val speedyObservers = SExpr.SEValue(SValue.SList(FrontStack.Empty))
+    val machine =
+      Speedy.Machine.fromUpdateSExpr(
+        pkgs,
+        transactionSeed = crypto.Hash.hashPrivateKey("TransactionVersionTest"),
+        updateSE = SEMakeClo(
+          Array(),
+          1,
+          SExpr.SELet1General(
+            SBFetchAny(speedyContractId, SEValue.None),
+            SExpr.SEScopeExercise(
```

Comment:
`SEScopeExercise` is used to ensure that a root context is present at the end of command processing (ow Speedy will exception).